src/test/scala/ai/privado/threatEngine/go/ThreatTests.scala

```diff
@@ -0,0 +1,237 @@
+        |Reference link: https://github.com/OWASP/owasp-mstg/blob/v1.4.0/Document/0x05d-Testing-Data-Storage.md#testing-local-storage-for-sensitive-data-mstg-storage-1-and-mstg-storage-2
+        |""".stripMargin,
+      PolicyThreatType.THREAT,
+      PolicyAction.DENY,
+      DataFlow(List(), SourceFilter(Option(true), ""), List()),
+      List("**"),
+      Map[String, String](),
+      Map[String, String](),
+      "",
+      Array[String]()
+    )
+
+    "When same data-element is part of multiple table in go file" in {
+
+      val threatEngine = code("""
+          |package models
+          |
+          |import (
+          |	"log"
+          |	"time"
+          |
+          |	"github.com/google/uuid"
+          |	"github.com/jinzhu/gorm"
+          |)
+          |
+          |// User Model
+          |type User struct {
+          |	// we need to use our own custom model because the id is different of the "gorm.Model"
+          |	ID      string `gorm:"PRIMARY KEY; UNIQUE" json:"id"`
+          |	Name    string `gorm:"type:varchar(255); NOT NULL" json:"name" validate:"required"`
+          |	Email   string `gorm:"type:varchar(255)" json:"email"`
+          |	Phone   string `gorm:"type:varchar(100); NOT NULL; UNIQUE; UNIQUE_INDEX" json:"phone" validate:"required"`
+          |	Address string `gorm:"type:text" json:"address"`
+          |	MyModel
+          |}
+          |
+          |// User Model
+          |type Student struct {
+          |	// we need to use our own custom model because the id is different of the "gorm.Model"
+          |	ID      string `gorm:"PRIMARY KEY; UNIQUE" json:"id"`
+          |	Name    string `gorm:"type:varchar(255); NOT NULL" json:"name" validate:"required"`
+          |	Email   string `gorm:"type:varchar(255)" json:"email"`
+          |	Phone   string `gorm:"type:varchar(100); NOT NULL; UNIQUE; UNIQUE_INDEX" json:"phone" validate:"required"`
+          |	Address string `gorm:"type:text" json:"address"`
+          |	MyModel
+          |}
+          |
+          |// Check if Users Model implements/match Model interface
+          |var _ Model = &User{}
+          |
+          |
+          |// FindOneByID gets a user by id
+          |func (u *User) FindOneByID(db *gorm.DB) (Model, error) {
+          |	err := db.Find(&u).Error
+          |	if err != nil {
+          |		return nil, err
+          |	}
+          |	return u, nil
+          |}
+          |
+          |
+          |// FindOneByID gets a user by id
+          |func (u *Student) FindOneByID(db *gorm.DB) (Model, error) {
+          |	err := db.Find(&u).Error
+          |	if err != nil {
+          |		return nil, err
+          |	}
+          |	return u, nil
+          |}
+          |
+          |""".stripMargin)
+
+      val result = threatEngine.processProcessingViolations(threat)
+      result should not be empty
+      result.get.policyId shouldBe "PrivadoPolicy.Storage.IsSamePIIShouldNotBePresentInMultipleTables"
```

Comment:
Do we also need to pin point what are Data Elements were part of different tables? 
cc- @dbMundada

Reply:
Added data-element as part of test case: https://github.com/Privado-Inc/privado-core/pull/814/commits/1f896170d831e6bfac21ffbb33348b16be85211c#diff-9b894724b8c80486fc9d176ab68ffe4e82cce2359fa4ff55b32c12abbda20d4dR126